### ISSUE TITLE
[BUG][PHP] x-group-parameters with bad/weird behavior

### ISSUE THREAD

**Issue by ypereirareis** [OpenAPITools/openapi-generator#2945]
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] What's the version of OpenAPI Generator used?
- [x] Have you search for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Bounty to sponsor the fix ([example](https://www.bountysource.com/issues/66123212-javascript-client-produces-a-wrong-object-for-a-string-enum-type-that-is-used-with-ref))

##### Description

A generated method is called with a list of parameters instead of a single parameter bag.

```php
public function getBookCollectionWithHttpInfo($associative_array)
{
    $request = $this->getBookCollectionRequest($associative_array['page'], $associative_array['items_per_page'], $associative_array['pagination']);
    ...
}

protected function getBookCollectionRequest($associative_array)
{
    // unbox the parameters from the associative array
    $page = array_key_exists('page', $associative_array) ? $associative_array['page'] : null;
    $items_per_page = array_key_exists('items_per_page', $associative_array) ? $associative_array['items_per_page'] : null;
    $pagination = array_key_exists('pagination', $associative_array) ? $associative_array['pagination'] : null;
    ...
}
```

EXPECTED 

```php
public function getBookCollectionWithHttpInfo($associative_array)
{
    $request = $this->getBookCollectionRequest($associative_array);
    ...
}

```

##### openapi-generator version

4.0.1-SNAPSHOT

##### OpenAPI declaration file content or url

```json
{
  "basePath": "/",
  "info": {
    "description": "Test API v3",
    "title": "TestAPI",
    "version": "1.0.7"
  },
  "paths": {
    "/api/book": {
      "get": {
        "operationId": "getBookCollection",
        "parameters": [
          {
            "description": "The collection page number",
            "in": "query",
            "name": "page",
            "required": false,
            "type": "integer"
          },
          {
            "description": "The number of items per page",
            "in": "query",
            "name": "itemsPerPage",
            "required": false,
            "type": "integer"
          },
          {
            "description": "Enable or disable pagination",
            "in": "query",
            "name": "pagination",
            "required": false,
            "type": "boolean"
          }
        ],
        "produces": [
          "application/ld+json",
          "application/json",
          "text/html"
        ],
        "responses": {
          "200": {
            "description": "bookcollection response",
            "schema": {
              "type": "string"
            }
          }
        },
        "summary": "",
        "tags": [
          "book"
        ],
        "x-group-parameters": true
      }
    }

  },
  "security": [
    {
      "apiKey": []
    }
  ],
  "securityDefinitions": {
    "apiKey": {
      "description": "Value for the X-AUTHENTICATION-TOKEN header",
      "in": "header",
      "name": "X-AUTHENTICATION-TOKEN",
      "type": "apiKey"
    }
  },
  "swagger": "2.0"
}

```

  -->

##### Command line used for generation

```
docker run --rm --user $(id -u):$(id -g) -v ${PWD}:/local \
    openapitools/openapi-generator-cli generate \
        --git-user-id "USER" \
        --git-repo-id "test-client-lib" \
        --release-note "Test Service Client Library" \
        -g php \
        -o /local/${SWAGGER_CLIENT_PATH} \
        -i /local/${SWAGGER_SERVICE}/config/packages/api_platform/swagger/api.json \
        -c /local/${SWAGGER_SERVICE}/config/packages/api_platform/swagger/generator.json
```

**Comment by wing328** [comment]
@ypereirareis Thanks for reporting the issue. I agree with you it's a bug. 

May I know if you've time to help submit a fix? I can show you a good starting point to fix the issue.

**Comment by ypereirareis** [comment]
Hi, and thanks for this awesome lib. :) 

Unfortunately, I have no time to work on a fix for now. I'm just using the generator to have my client. I haven't used the "x-group-parameters" option before. But I need it to avoid BC in my client lib when I add optional parameters to my endpoints/Apis.

What I can tell you if it can help, is that the same method but "Async" in the same file (api.mustache) is generated properly with correct parameter array, when its called.

Thanks.

**Comment by wing328** [comment]
That's ok. Thanks for the additional info, which definitely helps.

Let's see if anyone from the community has time to work on a fix for this.

**Comment by wing328** [comment]
@ypereirareis thanks for the hint, which helps me to fix the issue pretty easily.

Can you help review the fix (#2951) when you've time?
